packit_service/worker/events/new_hotness.py

```diff
@@ -0,0 +1,102 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+from logging import getLogger
+from typing import Optional, Tuple
+from urllib.parse import urlparse
+
+from packit_service.models import ProjectReleaseModel, AbstractTriggerDbType
+from packit_service.worker.events import AbstractForgeIndependentEvent
+
+logger = getLogger(__name__)
+
+
+class NewHotnessUpdateEvent(AbstractForgeIndependentEvent):
+    def __init__(
+        self,
+        package_name: str,
+        version: str,
+        distgit_project_url: str,
+    ):
+        super().__init__()
+        self.package_name = package_name
+        self.version = version
+        self.project_url = distgit_project_url
+
+        self._upstream_repo_namespace_and_name: Optional[Tuple] = None
+
+    @property
+    def upstream_project_url(self) -> str:
+        return self.package_config.upstream_project_url
+
+    def get_repo_namespace_and_name(self) -> Optional[Tuple]:
+        path = urlparse(self.upstream_project_url).path
+        if not path:
+            return None, None
+        # remove leading /
+        path = path[1:]
+        path_split = path.rsplit("/", maxsplit=1)
+        if len(path_split) != 2:
+            return None, None
+        return path_split[0], path_split[1]
+
+    @property
+    def upstream_repo_namespace_and_name(self) -> Optional[Tuple]:
+        if not self._upstream_repo_namespace_and_name:
+            self._upstream_repo_namespace_and_name = self.get_repo_namespace_and_name()
+        return self._upstream_repo_namespace_and_name
+
+    @property
+    def upstream_repo_namespace(self) -> Optional[str]:
+        return self.upstream_repo_namespace_and_name[0]
+
+    @property
+    def upstream_repo_name(self) -> Optional[str]:
+        return self.upstream_repo_namespace_and_name[1]
+
+    @property
+    def tag_name(self):
+        if not self.package_config.upstream_tag_template:
+            return self.version
+
+        return self.package_config.upstream_tag_template.format(version=self.version)
+
+    @property
+    def db_trigger(self) -> Optional[AbstractTriggerDbType]:
+        return ProjectReleaseModel.get_or_create(
+            tag_name=self.tag_name,
+            namespace=self.upstream_repo_namespace,
+            repo_name=self.upstream_repo_name,
+            project_url=self.upstream_project_url,
+            # TODO do we need to have the upstream commit hash?
+            commit_hash=None,
+        )
+
+    @property
+    def commit_sha(self):
+        # this is used for getting the package configuration, by default the method
+        # gets it from the default branch of the repository which is what we want
+        return None
+
+    def pre_check(self):
+        if not self.package_config:
```

Comment:
Could you please add a docstring here to sum up what exactly is being checked? :)